### PR TITLE
fix: beagle helpers page and add beagle grpc tutorials

### PR DESCRIPTION
**Pull request by luismaximozup**

### CODE REVIEW

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+---
+title: "Beagle gRPC for Web"
+weight: 1
+description: >-
+  In this section you will find all the info related to the client and the CLI of gRPC for Beagle Web (Angular and React).
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:5]:
```suggestion
  In this section, you will find information about the client and the CLI of gRPC for Beagle Web (Angular and React).
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+## What is gRPC?
+
+---
+
+It is an open source framework to perform RPC (Remote call procedure) calls, through the HTTP2 protocol in a two-way streaming format, using Protobufs as interfaces between client and server.
+
+Protobufs, or Protocol Buffers, are a structured data serialization mechanism created by Google, which aim to keep contracts always valid and consistent regarding the implementations that consume a gRPC service.
+
+Proto files, in their content, are very similar to any interface we are used to on the daily basis, but with the .proto extension, where we define which objects and methods are likely to be used in the service's communication.
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:16]:
```suggestion
- **Protobufs or Protocol Buffers**: They are structured data serialization mechanisms created by Google, which aim to keep contracts always valid and consistent, regarding the implementations that consume a gRPC service.

- **Proto files**: They are very similar to any interface we are used to on the daily basis, in their content. However, they have the .proto extension, where you can define which objects and methods are likely to be used in the service's communication.
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+[Learn more...](https://grpc.io/)
+
+## gRPC for Beagle Web
+
+---
+
+gRPC is a technology that makes use of the **HTTP 2** protocol, however, this technology is not compatible with the current Web structure, so the structure created uses a Client and a CLI, where the CLI is responsible for handling gRPC connections through a Proxy, written with Go, and the Client is responsible for handling all requests made by Beagle to the Proxy.
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:24]:
```suggestion
gRPC is a technology that uses the **HTTP 2** protocol, however, this technology is not compatible with the current Web structure. So, the new structure created uses a Client and a CLI: 
- The CLI is responsible for handling gRPC connections through a Proxy written with Go
- The Client is responsible for handling all requests made by Beagle to the Proxy.
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+- [Git - CLI](https://git-scm.com/): Used internally as a proxy dependency.
+- [Go - CLI](https://golang.org/): Language which the proxy is written.
+- [Dep - CLI](https://github.com/golang/dep): Go's dependency manager, where it checks that everything is installed and working as expected.
+- [Protoc - CLI](https://github.com/protocolbuffers/protobuf): Used by protobufs and necessary for code generation through a `.proto` file.
+
+## Installation
+
+---
+
+To install the library, install it globally, for a better experience with the CLI, and then install it in your project using the following commands:
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:39]:
```suggestion
Install the library globally. For a better experience with the CLI, install it in your project using the following commands:
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+yarn global add @zup-it/beagle-web-grpc
+yarn add --dev @zup-it/beagle-web-grpc
+```
+
+## Getting Started
+
+---
+
+1. After installing the library, in the project and globally, create the configuration file in the project's root folder, using: `beagle-web-grpc init` and then configure the attributes according to your needs.
+2. Then import the gRPC Client:
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:58]:
```suggestion
To start using, follow the next steps: 
**Step 1.** After installing the library in the project and globally, create the configuration file in the project's root folder, using: **`beagle-web-grpc init`** and then configure the attributes according to your needs.
**Step 2.**  Import the gRPC Client, check out below: 
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+       components: {},
+       fetchData: usingBeagleGrpcClient({
+         proxyAddress: "http://localhost:8081",
+       }),
+     })
+     export class Beagle {}
+     ```
+
+3. Start your proxy running the following command on the root of the project: `beagle-web-grpc start-proxy` or `beagle-web-grpc spx` or `beagle-web-grpc start-proxy --mode your-mode`.
+4. Use the Beagle Web as usually.
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:96]:
```suggestion
**Step 3.** Start your proxy running the following command on the root of the project: **`beagle-web-grpc start-proxy`** or **`beagle-web-grpc spx`** or **`beagle-web-grpc start-proxy --mode your-mode`.**
**Step 4.** Use the Beagle Web as usually.
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+> **Observations**
+> Make sure that your BFF Application is running.
+
+## CLI
+
+---
+
+As gRPC is not compatible with the current structure of the Web, to support the technology, it is necessary to use a Proxy to establish the communication between the REST application and the gRPC Backend application, so we created a CLI (command-line interface), **beagle-web-grpc**, to facilitate the use of the proxy and provide all the necessary settings.
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:105]:
```suggestion
gRPC is not compatible with the current structure of the Web, to support the technology, you have to use a Proxy to establish the communication between the REST application and the gRPC Backend application. 
Beagle team created a CLI (command-line interface), **beagle-web-grpc**, to make it easier to use the proxy and to provide all the necessary settings.
```

File: content/en/resources/gRPC/beagle-grpc-web.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+| `grpcBackendAddress` | Defines the address of the backend server that communicates using gRPC. Remembering that all requests that match this route will be redirected to the Proxy |
+| `tlsCertificatePath` | Set path to TLS Certificate location                                                                                                                        |
+| `tlsKeyPath`         | Set path to TLS Key location                                                                                                                                |
+| `runProxyOnPort`     | Defines the Port on which the gRPC Proxy will run                                                                                                           |
+
+## Client
+
+---
+
+The gRPC Client for Beagle Web is what redirects grpc requests to the proxy, the client is used through `BeagleService` in React and the `Beagle` module in Angular, using the `fetchData` property, which Beagle allows you to create custom HTTP requests, so the use goes like this:
```
Comment (jessica-tw) [inline content/en/resources/gRPC/beagle-grpc-web.md:153]:
```suggestion
The gRPC Client for Beagle Web is what redirects gRPC requests to the proxy. The client is used through `BeagleService` in React and `Beagle`'s module in Angular. When using the `fetchData` property, Beagle allows you to create custom HTTP requests, so it goes like this:
```

File: content/pt/resources/grpc/_index.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+---
+title: gRPC
+weight: 2
+description: >-
+  Esta seção contem algumas bibliotecas que irão te auxiliar a usar o Beagle com gRPC.
```
Comment (jessica-tw) [inline content/pt/resources/grpc/_index.md:5]:
```suggestion
  Esta seção contém algumas bibliotecas ajudam você a usar o Beagle com gRPC.
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+
+## O que é gRPC?
+
+---
+
+É um framework open source para realizar chamadas RPC (Remote call procedure), através do protocolo HTTP2 em formato de two-way streaming, utilizando Protobufs como interfaces entre cliente e servidor.
+
+Protobufs ou Protocol Buffers, são um mecanismo de serialization de dados estruturados criados pelo Google, onde visam manter os contratos sempre válidos e coerentes quanto às implementações que consomem aquele serviço gRPC.
+
+Arquivos proto, em seu conteúdo, são bem parecidos com qualquer interface que estamos acostumados no dia-a-dia, porém com a extensão .proto, onde nele definimos quais são os objetos e métodos que são passíveis de serem usados na comunicação do serviço.
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:16]:
```suggestion
É um framework open source que realiza chamadas RPC (Remote call procedure) por meio do protocolo HTTP2 em formato de two-way streaming, utilizando Protobufs como interfaces entre cliente e servidor.

- Protobufs ou Protocol Buffers: São mecanismos de serialização de dados estruturados criados pelo Google, que visam manter os contratos sempre válidos e coerentes quanto às implementações que consomem aquele serviço gRPC.

- Arquivos proto:  São bem parecidos com qualquer interface que você está acostumado no dia-a-dia, em seu conteúdo. Porém com a extensão .proto, onde nele você define quais são os objetos e métodos que podem ser usados na comunicação do serviço.
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+[Saiba mais...](https://grpc.io/)
+
+## gRPC para Beagle Web
+
+---
+
+O gRPC é uma tecnologia que faz uso do protocolo **HTTP 2**, todavia, esta tecnologia não é compatível com a atual estrutura da Web, por isso a estrutura criada faz uso de um Cliente e um CLI, onde o CLI fica responsável por manipular as conexões gRPC através de um Proxy, em Go, e o Cliente fica responsável por manipular todas requisições feitas pelo Beagle para o Proxy.
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:24]:
```suggestion
O gRPC é uma tecnologia que utiliza o protocolo **HTTP 2**, no entanto essa tecnologia não é compatível com a atual estrutura da Web. Por isso, a estrutura criada faz uso de um Cliente e um CLI:
- O CLI fica responsável por manipular as conexões gRPC através de um Proxy em Go; 
- O Cliente fica responsável por manipular todas requisições feitas pelo Beagle para o Proxy.
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+- [Git - CLI](https://git-scm.com/): Usado internamente para as dependências do proxy.
+- [Go - CLI](https://golang.org/): Linguagem usada no proxy.
+- [Dep - CLI](https://github.com/golang/dep): Controlador de dependências do Go, onde verifica se tudo está instalado como esperado.
+- [Protoc - CLI](https://github.com/protocolbuffers/protobuf): Utilizado pelos protobufs e necessário para geração de código através de um arquivo `.proto`.
+
+## Instalação
+
+---
+
+Para instalar a biblioteca, instale ela globalmente, para melhor experiência com o CLI e após instale a mesma em seu projeto com os seguintes comandos:
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:39]:
```suggestion
Instale a biblioteca globalmente. Para melhorar a experiência com o CLI, instale a mesma em seu projeto com os seguintes comandos:
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+yarn global add @zup-it/beagle-web-grpc
+yarn add --dev @zup-it/beagle-web-grpc
+```
+
+## Utilização
+
+---
+
+1. Após a instalação das bibliotecas, crie o arquivo de configurações na pasta raiz do projeto, utilizando: `beagle-web-grpc init` e configure os atributos conforme as suas necessidades.
+2. Importe o cliente gRPC:
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:58]:
```suggestion
Para começar a utilizar, siga os passos abaixo:

**Passo 1.** Após a instalação das bibliotecas, crie o arquivo de configurações na pasta raiz do projeto, utilizando **`beagle-web-grpc init`** e configure os atributos com o que você precisa.
**Passo 2.** Importe o cliente gRPC, veja abaixo:
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+       components: {},
+       fetchData: usingBeagleGrpcClient({
+         proxyAddress: "http://localhost:8081",
+       }),
+     })
+     export class Beagle {}
+     ```
+
+3. Inicie o Proxy gRPC utilizando: `beagle-web-grpc start-proxy` ou `beagle-web-grpc start-proxy --mode seu-modo-personalizado`.
+4. Utilize o Beagle normalmente.
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:96]:
```suggestion
**Passo 3.** Inicie o Proxy gRPC utilizando: **`beagle-web-grpc start-proxy`** ou **`beagle-web-grpc start-proxy --mode seu-modo-personalizado`**.
**Passo 4.** Utilize o Beagle normalmente.
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+
+> **Observação**
+> Tenha sua aplicação BFF rodando ao mesmo tempo.
+
+## CLI
+
+---
+
+Como o gRPC não é compatível com a estrutura atual da Web, para que a tecnologia tenha suporte, se faz necessário o uso de um Proxy para fazer a comunicação entre REST e o Backend gRPC, por isso criamos um CLI (Command-line Interface), **beagle-web-grpc**, para que facilite o uso do proxy e provenha todas as configurações necessárias.
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:105]:
```suggestion
O gRPC não é compatível com a estrutura atual da Web, para que a tecnologia tenha suporte, você precisa usar um Proxy para fazer a comunicação entre REST e o Backend gRPC. 
Por isso o time do Beagle criou um CLI (Command-line Interface), **beagle-web-grpc**, para facilitar o uso do proxy e  adicionar todas as configurações necessárias.
```

File: content/pt/resources/grpc/beagle-grpc-web.md [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+| `grpcBackendAddress` | Define o endereço do servidor backend que se comunica utilizando gRPC. Lembrando que todas as requisições que baterem com esta rota, serão redirecionadas para o Proxy |
+| `tlsCertificatePath` | Define o caminho para o local do Certificado TLS                                                                                                                       |
+| `tlsKeyPath`         | Define o caminho para o local da Chave TLS                                                                                                                             |
+| `runProxyOnPort`     | Define a Porta em que o Proxy gRPC será executado                                                                                                                      |
+
+## Cliente
+
+---
+
+O Cliente gRPC para Beagle Web é o que faz o redirecionamento das requisições grpc para o proxy, o cliente é utilizado através do `BeagleService` no React e do módulo `Beagle` no Angular, se valendo da propriedade de `fetchData`, ao qual o Beagle permite criar requisições HTTP customizadas, então a utilização se dá desta maneira:
```
Comment (jessica-tw) [inline content/pt/resources/grpc/beagle-grpc-web.md:153]:
```suggestion
O Cliente gRPC para Beagle Web faz o redirecionamento das requisições gRPC para o proxy. O cliente é utilizado por meio do `BeagleService` no React e do módulo `Beagle` no Angular. 
Se valendo da propriedade de `fetchData` em que o Beagle permite criar requisições HTTP customizadas, veja abaixo como a utilização será:
```

**Review by jessica-tw — COMMENTED** [review]
TW review done ;)

**Review by arthurbleilzup — DISMISSED** [review]

**Review by arthurbleilzup — APPROVED** [review]

**Review by paulomeurerzup — CHANGES_REQUESTED** [review]

**Review by paulomeurerzup — APPROVED** [review]

**Pull request merged**